cdc/kv/region_worker.go

```diff
@@ -683,17 +683,20 @@ func (w *regionWorker) handleEventEntry(
 			state.matcher.putPrewriteRow(entry)
 		case cdcpb.Event_COMMIT:
 			w.metrics.metricPullEventCommitCounter.Inc()
+			// NOTE: state.getLastResolvedTs() will never less than session.startTs.
 			resolvedTs := state.getLastResolvedTs()
-			if entry.CommitTs <= resolvedTs {
+			// TiKV can send events with StartTs/CommitTs less than startTs.
+			isStaleEvent := entry.CommitTs <= w.session.startTs
```

Comment:
Should StaleEvents be filtered outside the switch code block?

Reply:
`state.matcher.matchRow` should always be called even if the event is stale.